### PR TITLE
Apply PHP 7.4 syntax and typed property

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 7.4, php 7.4 syntax can be applied with typed properties. 

- For `final class`, typed properties for protected is allowed if no inherit of parent or used by its final class parent.
- Otherwise, update only private properties.

### CODE REVIEW

**Comment by samsonasik** [comment]
All green 🎉

File: src/UndisclosedPassword.php [hunk 1]
```diff
@@ -39,13 +39,9 @@ final class UndisclosedPassword extends AbstractValidator
         self::NOT_A_STRING      => 'The provided password is not a string, please provide a correct password',
     ];
 
-    // phpcs:enable
-
-    /** @var ClientInterface */
-    private $httpClient;
+    private ClientInterface $httpClient;
```
Comment (andrew-demb) [inline src/UndisclosedPassword.php:42]:
```suggestion
    // phpcs:enable

    private ClientInterface $httpClient;
```

**Review by Ocramius — APPROVED** [review]
Thanks @samsonasik !

File: src/UndisclosedPassword.php (hunk 1 above)
Comment (Ocramius) [reply src/UndisclosedPassword.php:42]: Ha, CS completely disabled here, didn't notice :O

File: test/CallbackTest.php [hunk 2]
```diff
@@ -90,9 +90,7 @@ public function testCanAcceptContextWithoutOptions(): void
     {
         $value     = 'bar';
         $context   = ['foo' => 'bar', 'bar' => 'baz'];
-        $validator = new Callback(function ($v, $c) use ($value, $context) {
-            return ($value === $v) && ($context === $c);
-        });
+        $validator = new Callback(static fn($v, $c) => ($value === $v) && ($context === $c));
```
Comment (holtkamp) [inline test/CallbackTest.php:93]: Why not use a return type ` : bool` here?

Comment (Ocramius) [reply test/CallbackTest.php:93]: @holtkamp I think that requires more additions to the setup that @samsonasik uses (he's a [Rector](https://github.com/rectorphp/rector) maintainer)

Comment (samsonasik) [reply test/CallbackTest.php:93]: Yes, that's can be handled by type declaration set list, it was missed cherry pick return type declaration rule to apply, though need to only apply to ArrowFunction

**Pull request merged**